### ISSUE TITLE
Can't launch OctoPrint after installation.

### ISSUE THREAD

**Issue by Nihilianth** [feelfreelinux/octo4a#23]
I've successfully installed the apk on my OnePlus 5 (Running Android 10).
However,  the installer gets stuck at "Booting OctoPrint for the first time…". Re-launching the application has no effect.

My device is not rooted atm. After some googling I've found this  [https://github.com/termux/termux-app/issues/299](issue) on Termux github. It seems to be related but I am wondering if there is perhaps a workaround that doesnt require rooting.

ADB log: 
```07-27 15:06:06.915 32246 32246 W octoprint: type=1400 audit(0.0:14517): avc: granted { execute } for name="clang-9" dev="sda13" ino=106565 scontext=u:r:untrusted_app_27:s0:c205,c257,c512,c768 tcontext=u:object_r:app_data_file:s0:c205,c257,c512,c768 tclass=file
07-27 15:06:06.925 32321 32321 W gcc     : type=1400 audit(0.0:14520): avc: granted { execute } for path="/data/data/com.octo4a/files/usr/bin/clang-9" dev="sda13" ino=106565 scontext=u:r:untrusted_app_27:s0:c205,c257,c512,c768 tcontext=u:object_r:app_data_file:s0:c205,c257,c512,c768 tclass=file
07-27 15:06:06.935 32321 32321 W gcc     : type=1400 audit(0.0:14521): avc: granted { execute } for path="/data/data/com.octo4a/files/usr/lib/libncursesw.so.6.1" dev="sda13" ino=121 scontext=u:r:untrusted_app_27:s0:c205,c257,c512,c768 tcontext=u:object_r:app_data_file:s0:c205,c257,c512,c768 tclass=file
07-27 15:06:07.945 32246 32246 W octoprint: type=1400 audit(0.0:14571): avc: granted { execute } for name="objdump" dev="sda13" ino=106719 scontext=u:r:untrusted_app_27:s0:c205,c257,c512,c768 tcontext=u:object_r:app_data_file:s0:c205,c257,c512,c768 tclass=file
07-27 15:06:07.955 32328 32328 W objdump : type=1400 audit(0.0:14575): avc: granted { execute } for path="/data/data/com.octo4a/files/usr/lib/libopcodes-2.33.1.so" dev="sda13" ino=97 scontext=u:r:untrusted_app_27:s0:c205,c257,c512,c768 tcontext=u:object_r:app_data_file:s0:c205,c257,c512,c768 tclass=file
07-27 15:06:08.246 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: 2021-07-27 15:06:08,241 - octoprint.plugin.core - INFO - Plugin Pi Support Plugin (2021.6.14) did not pass check, not loading.
07-27 15:06:08.351 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: 2021-07-27 15:06:08,347 - octoprint.plugin.core - INFO - Found 16 plugin(s) providing 16 mixin implementations, 32 hook handlers
07-27 15:06:08.408 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: 2021-07-27 15:06:08,403 - octoprint.server.heartbeat - INFO - Starting server heartbeat, 900.0s interval
07-27 15:06:08.435 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: Traceback (most recent call last):
07-27 15:06:08.437 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/psutil/_common.py", line 403, in wrapper
07-27 15:06:08.439 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     return cache[key]
07-27 15:06:08.451 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: KeyError: (('/proc',), frozenset())
07-27 15:06:08.452 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:
07-27 15:06:08.453 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: During handling of the above exception, another exception occurred:
07-27 15:06:08.454 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:
07-27 15:06:08.455 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: Traceback (most recent call last):
07-27 15:06:08.456 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/psutil/_pslinux.py", line 302, in <module>
07-27 15:06:08.456 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     set_scputimes_ntuple("/proc")
07-27 15:06:08.457 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/psutil/_common.py", line 405, in wrapper
07-27 15:06:08.458 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     ret = cache[key] = fun(*args, **kwargs)
07-27 15:06:08.458 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/psutil/_pslinux.py", line 269, in set_scputimes_ntuple
07-27 15:06:08.459 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     with open_binary('%s/stat' % procfs_path) as f:
07-27 15:06:08.460 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/psutil/_common.py", line 711, in open_binary
07-27 15:06:08.461 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     return open(fname, "rb", **kwargs)
07-27 15:06:08.462 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: PermissionError: [Errno 13] Permission denied: '/proc/stat'
07-27 15:06:08.476 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: 2021-07-27 15:06:08,433 - octoprint.cli.server - ERROR - Uncaught exception
07-27 15:06:08.476 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: Traceback (most recent call last):
07-27 15:06:08.476 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/bin/octoprint", line 11, in <module>
07-27 15:06:08.476 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     load_entry_point('OctoPrint==1.6.1', 'console_scripts', 'octoprint')()
07-27 15:06:08.477 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/octoprint/__init__.py", line 948, in main
07-27 15:06:08.477 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     octo(args=args, prog_name="octoprint", auto_envvar_prefix="OCTOPRINT")
07-27 15:06:08.477 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/click/core.py", line 829, in __call__
07-27 15:06:08.477 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     return self.main(*args, **kwargs)
07-27 15:06:08.477 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/click/core.py", line 782, in main
07-27 15:06:08.478 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     rv = self.invoke(ctx)
07-27 15:06:08.478 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/click/core.py", line 1236, in invoke
07-27 15:06:08.478 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     return Command.invoke(self, ctx)
07-27 15:06:08.478 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/click/core.py", line 1066, in invoke
07-27 15:06:08.478 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     return ctx.invoke(self.callback, **ctx.params)
07-27 15:06:08.479 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/click/core.py", line 610, in invoke
07-27 15:06:08.479 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     return callback(*args, **kwargs)
07-27 15:06:08.479 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
07-27 15:06:08.482 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     return f(get_current_context(), *args, **kwargs)
07-27 15:06:08.483 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/octoprint/cli/__init__.py", line 348, in octo
07-27 15:06:08.483 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     ctx.invoke(serve_command, **kwargs)
07-27 15:06:08.483 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/click/core.py", line 610, in invoke
07-27 15:06:08.483 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     return callback(*args, **kwargs)
07-27 15:06:08.484 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/click/decorators.py", line 21, in new_func
07-27 15:06:08.484 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     return f(get_current_context(), *args, **kwargs)
07-27 15:06:08.484 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/octoprint/cli/server.py", line 317, in serve_command
07-27 15:06:08.484 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     run_server(
07-27 15:06:08.484 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/octoprint/cli/server.py", line 164, in run_server
07-27 15:06:08.484 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     octoprint_server.run()
07-27 15:06:08.485 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/octoprint/server/__init__.py", line 371, in run
07-27 15:06:08.485 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     self._start_intermediary_server()
07-27 15:06:08.485 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/site-packages/octoprint/server/__init__.py", line 2481, in _start_intermediary_server
07-27 15:06:08.485 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     self._intermediary_server.server_bind()
07-27 15:06:08.485 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/http/server.py", line 137, in server_bind
07-27 15:06:08.486 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     socketserver.TCPServer.server_bind(self)
07-27 15:06:08.486 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:   File "/data/data/com.octo4a/files/usr/lib/python3.8/socketserver.py", line 466, in server_bind
07-27 15:06:08.486 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint:     self.socket.bind(self.server_address)
07-27 15:06:08.486 28927 32247 D OctoPrintHandlerRepositoryImpl: : octoprint: OSError: [Errno 98] Address already in use
```

**Comment by feelfreelinux** [comment]
Corrupted installations detection fixed in 1.0.0 https://github.com/feelfreelinux/octo4a/releases/tag/1.0.0

Please reopen the issue if you can still reproduce it.

**Comment by metalichesky** [comment]
I faced the same problem in octo4a version 1.1.1. I am using Ugoose X3 Pro device, Android 9. 
Steps to reproduce: 
- Install the octo4a app. 
- Run the app for first time, installing of octoprint started
- At the end of installation, the application crashes. When you try to run it again, in the application logs the following output:
 
```
🐸 % Total % Received % Xferd Average Speed Time Time Time Current
🐸 Dload Upload Total Spent Left Speed
🐸
🐸 0 0 0 0 0 0 0 0 —:--:— —:--:— —:--:— 0
🐸 100 3675 100 3675 0 0 6871 0 —:--:— —:--:— —:--:— 6882
🔌 Got extension Remote web terminal (ttyd)
🐙 2022-08-19 06:31:11,961 - octoprint.startup - INFO - ****************************************************************************** [0m
🐙 2022-08-19 06:31:11,964 - octoprint.startup - INFO - Starting OctoPrint 1.8.2 [0m
🐙 2022-08-19 06:31:11,965 - octoprint.startup - INFO - ****************************************************************************** [0m
🐙 2022-08-19 06:31:12,054 - octoprint.util.connectivity.connectivity_checker - INFO - Connectivity state is currently: online [0m
🐙 2022-08-19 06:31:12,055 - octoprint.util.connectivity.connectivity_checker - INFO - Connecting to 1.1.1.1:53 is working [0m
🐙 2022-08-19 06:31:12,055 - octoprint.util.connectivity.connectivity_checker - INFO - Resolving [octoprint.org](https://vk.com/away.php?to=http%3A%2F%2Foctoprint.org&cc_key=) is working [0m
🐙 2022-08-19 06:31:12,114 - octoprint.plugin.core - INFO - Loading plugins from /usr/lib/python3.9/site-packages/octoprint/plugins, /root/.octoprint/plugins and installed plugin packages... [0m
🐙 2022-08-19 06:31:17,197 - octoprint.plugin.core - INFO - Plugin Pi Support Plugin (2022.6.13) did not pass check, not loading. [0m
🐙 2022-08-19 06:31:17,743 - octoprint.plugin.core - INFO - Found 18 plugin(s) providing 18 mixin implementations, 34 hook handlers [0m
🐙 2022-08-19 06:31:17,892 - octoprint.server.heartbeat - INFO - Starting server heartbeat, 900.0s interval [0m
🐙 [31m2022-08-19 06:31:17,983 - octoprint.cli.server - ERROR - Uncaught exception [0m
🐙 Traceback (most recent call last):
🐙 File "/usr/bin/octoprint", line 33, in <module>
🐙 sys.exit(load_entry_point('OctoPrint==1.8.2', 'console_scripts', 'octoprint')())
🐙 File "/usr/lib/python3.9/site-packages/octoprint/[__init__.py](https://vk.com/away.php?to=http%3A%2F%2F__init__.py&cc_key=)", line 936, in main
🐙 octo(args=args, prog_name="octoprint", auto_envvar_prefix="OCTOPRINT")
🐙 File "/usr/lib/python3.9/site-packages/click/[core.py](https://vk.com/away.php?to=http%3A%2F%2Fcore.py&cc_key=)", line 1130, in __call__
🐙 return self.main(*args, **kwargs)
🐙 File "/usr/lib/python3.9/site-packages/click/[core.py](https://vk.com/away.php?to=http%3A%2F%2Fcore.py&cc_key=)", line 1055, in main
🐙 rv = self.invoke(ctx)
🐙 File "/usr/lib/python3.9/site-packages/click/[core.py](https://vk.com/away.php?to=http%3A%2F%2Fcore.py&cc_key=)", line 1657, in invoke
🐙 return _process_result(sub_ctx.command.invoke(sub_ctx))
🐙 File "/usr/lib/python3.9/site-packages/click/[core.py](https://vk.com/away.php?to=http%3A%2F%2Fcore.py&cc_key=)", line 1404, in invoke
🐙 return ctx.invoke(self.callback, **ctx.params)
🐙 File "/usr/lib/python3.9/site-packages/click/[core.py](https://vk.com/away.php?to=http%3A%2F%2Fcore.py&cc_key=)", line 760, in invoke
🐙 return __callback(*args, **kwargs)
🐙 File "/usr/lib/python3.9/site-packages/click/[decorators.py](https://vk.com/away.php?to=http%3A%2F%2Fdecorators.py&cc_key=)", line 26, in new_func
🐙 return f(get_current_context(), *args, **kwargs)
🐙 File "/usr/lib/python3.9/site-packages/octoprint/cli/[server.py](https://vk.com/away.php?to=http%3A%2F%2Fserver.py&cc_key=)", line 317, in serve_command
🐙 run_server(
🐙 File "/usr/lib/python3.9/site-packages/octoprint/cli/[server.py](https://vk.com/away.php?to=http%3A%2F%2Fserver.py&cc_key=)", line 163, in run_server
🐙 [octoprint_server.run](https://vk.com/away.php?to=http%3A%2F%2Foctoprint_server.run&cc_key=)()
🐙 File "/usr/lib/python3.9/site-packages/octoprint/server/[__init__.py](https://vk.com/away.php?to=http%3A%2F%2F__init__.py&cc_key=)", line 369, in run
🐙 self._start_intermediary_server()
🐙 File "/usr/lib/python3.9/site-packages/octoprint/server/[__init__.py](https://vk.com/away.php?to=http%3A%2F%2F__init__.py&cc_key=)", line 2500, in _start_intermediary_server
🐙 self._intermediary_server.server_bind()
🐙 File "/usr/lib/python3.9/http/[server.py](https://vk.com/away.php?to=http%3A%2F%2Fserver.py&cc_key=)", line 138, in server_bind
🐙 socketserver.TCPServer.server_bind(self)
🐙 File "/usr/lib/python3.9/[socketserver.py](https://vk.com/away.php?to=http%3A%2F%2Fsocketserver.py&cc_key=)", line 466, in server_bind
🐙 self.socket.bind(self.server_address)
🐙 OSError: [Errno 98] Address in use
```
Unfortunately, I did not manage to get the log of the crash itself from logcat
